### PR TITLE
Cover btp package with tests

### PR DESCRIPTION
**Pull request by pPrecel**
**Description**

Changes proposed in this pull request:

- test the `btp` package

**Related issue(s)**

- https://github.com/kyma-project/cli/issues/2085

### CODE REVIEW

File: internal/kube/btp/service_test.go [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+		)
+
+		serviceInstance, err := btpClient.GetServiceInstance(context.Background(), "test-namespace", "test-name")
+		require.ErrorContains(t, err, "serviceinstances.services.cloud.sap.com \"test-name\" not found")
+		require.Nil(t, serviceInstance)
+	})
+}
+
+func Test_btpClient_CreateServiceInstance(t *testing.T) {
+	t.Run("get ServiceInstance", func(t *testing.T) {
```
Comment (anoipm) [inline internal/kube/btp/service_test.go:43]: Is it proper name for this test?

File: internal/kube/btp/service_test.go [hunk 2]
```diff
@@ -0,0 +1,360 @@
...
+		)
+
+		serviceBinding, err := btpClient.GetServiceBinding(context.Background(), "test-namespace", "test-name")
+		require.ErrorContains(t, err, "servicebindings.services.cloud.sap.com \"test-name\" not found")
+		require.Nil(t, serviceBinding)
+	})
+}
+
+func Test_btpClient_CreateServiceBinding(t *testing.T) {
+	t.Run("get ServiceBinding", func(t *testing.T) {
```
Comment (anoipm) [inline internal/kube/btp/service_test.go:118]: Is it proper name?

File: internal/kube/btp/service_test.go [hunk 3]
```diff
@@ -0,0 +1,360 @@
...
+		givenBinding.Object["status"] = fixUnstructuredFailedCommonStatus()
+
+		dynamic := dynamic_fake.NewSimpleDynamicClient(scheme, givenBinding)
+		btpClient := NewClient(
+			dynamic,
+		)
+
+		readyFn := btpClient.IsBindingReady(context.Background(), "test-namespace", "test-name")
+		done, err := readyFn(context.Background())
+		require.Error(t, err)
```
Comment (anoipm) [inline internal/kube/btp/service_test.go:195]: Could we check which error was returned?

File: internal/kube/btp/service_test.go [hunk 4]
```diff
@@ -0,0 +1,360 @@
...
+		givenInstance.Object["status"] = fixUnstructuredFailedCommonStatus()
+
+		dynamic := dynamic_fake.NewSimpleDynamicClient(scheme, givenInstance)
+		btpClient := NewClient(
+			dynamic,
+		)
+
+		readyFn := btpClient.IsInstanceReady(context.Background(), "test-namespace", "test-name")
+		done, err := readyFn(context.Background())
+		require.Error(t, err)
```
Comment (anoipm) [inline internal/kube/btp/service_test.go:246]: Could we check this error?

File: internal/kube/btp/service_test.go [hunk 5]
```diff
@@ -0,0 +1,360 @@
...
+		)
+
+		readyFn := btpClient.IsInstanceReady(context.Background(), "test-namespace", "test-name")
+		done, err := readyFn(context.Background())
+		require.ErrorContains(t, err, "serviceinstances.services.cloud.sap.com \"test-name\" not found")
+		require.False(t, done)
+	})
+}
+
+func fixServiceInstance() *ServiceInstance {
```
Comment (anoipm) [inline internal/kube/btp/service_test.go:266]: Now we have corresponding methods `fix...` and `fixUnstructured...`. Both versions create very similar structures.
Is it possible to create only one structure and convert (to/from unstructured) it in second method?

Comment (pPrecel) [reply internal/kube/btp/service_test.go]: good point

**Review by anoipm — APPROVED** [review]

**Pull request merged**